src/analysis/retail/warlock/demonology/modules/talents/SoulStrike.tsx

```diff
@@ -25,24 +28,56 @@ class SoulStrike extends Analyzer {
       Events.damage.by(SELECTED_PLAYER_PET).spell(SPELLS.SOUL_STRIKE_DAMAGE),
       this.handleSoulStrikeDamage,
     );
+    this.addEventListener(
+      Events.resourcechange.by(SELECTED_PLAYER_PET).spell(SPELLS.SOUL_STRIKE_SHARD_GEN),
+      this.handleSoulStrikeShardGen,
+    );
   }
 
   handleSoulStrikeDamage(event: DamageEvent) {
     this.damage += event.amount + (event.absorbed || 0);
   }
 
+  handleSoulStrikeShardGen(event: ResourceChangeEvent) {
+    if (event.resourceChangeType === RESOURCE_TYPES.SOUL_SHARDS.id) {
+      this.totalCasts += 1;
```

Comment:
Shouldn't this entire function listen to the getAdjustedGain function which returns mathed out gain and waste, which you can then use in here?